### ISSUE TITLE
Mapping Property Object Over Target

### ISSUE THREAD

**Issue by SlowLogicBoy** [agileobjects/AgileMapper#64]
Is there a better way to do this?:
```csharp
class Statistics
{
    public float Ranking {get; set;}
    public string SomeOtherRankingStuff {get; set;}
}
class Model
{
   public string SomeOtherProperties {get;set;}
   public Statistics Statistics {get; set;}
}
class ModelDto
{
   public string SomeOtherProperties {get;set;}
   public float Ranking {get; set;}
   public string SomeOtherRankingStuff {get; set;}
}
Mapper.WhenMapping
    .From<Model>()
    .To<ModelDto>()
    .After.MappingEnds
    .Call(ctx => ctx.Source.Statistics.Map().Over(ctx.Target));
```

I want to map Statistics property Over ModelDto, since it doesn't have a seperate property for statistics?

**Comment by SteveWilkes** [comment]
Hi!

That's probably as good a way of doing it as exists currently - is there a configuration syntax or way of performing the mapping you'd like to be able to use?

Depending on your configuration-placement preferences, it's a tiny bit neater configuring it inline, imo:

```C#
var modelDto = model.Map().ToANew<ModelDto>(cfg => cfg
    .After.MappingEnds
    .Call((m, dto) => m.Statistics.Map().Over(dto)));
```

**Comment by SlowLogicBoy** [comment]
Hi,

Thanks for a fast response.

I was hoping to find something like
```csharp
cfg => cfg.WhenMapping.From<Video>()
    .To<YoutubeVideo>()
    .Map(ctx => ctx.Source.Statistics).ToSelf()
```
or maybe `.Map(ctx => ctx.Source.Statistics).To(t => t)` but this might be difficult since arguments is an expression. Or as a worst case (probably) `.Manual(Action<TTarget, TSource> manualFunc)`

**Comment by SteveWilkes** [comment]
I like `ToSelf()`, or maybe `ToTarget()` - I'll look at adding it in this week.

Thanks for the feedback! :)

**Comment by SteveWilkes** [comment]
Added in the latest code - release to follow:

```C#
Mapper.WhenMapping
    .From<Video>().To<YoutubeVideo>()
    .Map(ctx => ctx.Source.Statistics)
    .ToRootTarget();
```

...or:

```C#
var youtubeVideo = video.Map()
    .ToANew<YoutubeVideo>(cfg => cfg
        .Map(ctx => ctx.Source.Statistics)
        .ToRootTarget());
```

**Comment by SteveWilkes** [comment]
Available in [v0.24](https://www.nuget.org/packages/AgileObjects.AgileMapper) - thanks for the feedback! :)
